gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/CodeableConceptMapper.java

```diff
@@ -1,25 +1,162 @@
 package uk.nhs.adaptors.pss.translator.mapper;
 
+import java.util.Objects;
+
+import org.apache.commons.lang3.StringUtils;
 import org.hl7.fhir.dstu3.model.CodeableConcept;
+import org.hl7.fhir.dstu3.model.Extension;
+import org.hl7.fhir.dstu3.model.IdType;
+import org.hl7.fhir.dstu3.model.StringType;
 import org.hl7.v3.CD;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
+import lombok.RequiredArgsConstructor;
+import uk.nhs.adaptors.connector.dao.SnomedCTDao;
+import uk.nhs.adaptors.connector.model.SnomedCTDescription;
+
 @Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
 public class CodeableConceptMapper {
     private static final String SNOMED_SYSTEM_CODE = "2.16.840.1.113883.2.1.3.2.4.15";
     private static final String SNOMED_SYSTEM = "http://snomed.info/sct";
+    private static final String EXTENSION_URL = "https://fhir.nhs.uk/STU3/StructureDefinition/Extension-coding-sctdescid";
+    private static final String EXTENSION_DESCRIPTION_ID_URL = "descriptionId";
+    private static final String EXTENSION_DESCRIPTION_DISPLAY_URL = "descriptionDisplay";
+    private static final String CONCEPT_PARTITION_SHORT = "00";
+    private static final String CONCEPT_PARTITION_LONG = "10";
+    private static final String DESCRIPTION_PARTITION_SHORT = "01";
+    private static final String DESCRIPTION_PARTITION_LONG = "11";
+    private static final int MAX_CHARACTERS_FROM_RIGHT = 3;
+    private static final int MIN_CHARACTERS_FROM_RIGHT = 1;
 
-    public CodeableConcept mapToCodeableConcept(CD codedData) {
+    @Autowired
+    private final SnomedCTDao snomedCTDao;
+
+    public CodeableConcept mapToCodeableConcept(CD codedData, boolean isMedicationResource) {
```

Comment:
I think it would be better to create a separate method for medication resource. There's only one case when isMedicationResource is true and you need to pass false each time - better to have 2 public methods.